### ISSUE TITLE
Getting Same product list when pegination

### ISSUE THREAD

**Issue by akshaykatale99** [wp-graphql/wp-graphql-woocommerce#447]
**Describe the bug**
I am trying to query the product list with sort by price and categoryId. but after I call the next page by adding endCursor it seems to be getting the same product on the next page. each pagination request is getting the same product list.

I am using this query: 

        query GetProducts($featured: Boolean, $attribute: String, $attributeTerm: String, $categoryId: Int, $search: String, $sort: [ProductsOrderbyInput], $after: String, $taxFilter: [ProductTaxonomyFilterInput], $type: ProductTypesEnum, $sellerId: ID, $shuffle: Boolean, $onlyActiveAuction: Boolean, $categoryIdIn: [Int], $first: Int) {
          products(where: {attributeTerm: $attributeTerm, attribute: $attribute, categoryId: $categoryId, search: $search, orderby: $sort, taxonomyFilter: {and: $taxFilter}, featured: $featured, type: $type, sellerId: $sellerId, shuffle: $shuffle, onlyActiveAuction: $onlyActiveAuction, categoryIdIn: $categoryIdIn}, after: $after, first: $first) {
            nodes {
              image {
                sourceUrl(size: THUMBNAIL)
              }
              date
              featured
              name
              onSale
              id
              databaseId
              ... on SimpleProduct {
                id
                name
                price
                salePrice
                onSale
                regularPrice
              }
              shortDescription(format: RAW)
              ... on VariableProduct {
                id
                name
                variations {
                  nodes {
                    price
                    salePrice
                    regularPrice
                    name
                    onSale
                  }
                }
                price
              }
              ... on AuctionProduct {
                id
                name
                databaseId
                price
                regularPrice
                startPrice
                reservedPrice
              }
              type
            }
            pageInfo {
              endCursor
              hasNextPage
            }
          }
        }

Using this query variables :

        {
          "sort": [{
            "field": "PRICE",
            "order": "ASC"
          }],
          "first": 20,
          "after": "YXJyYXljb25uZWN0aW9uOjM0MTk4",
          "categoryId": 277
        }

Using this query, whenever I call the second page by adding endCursor I'm getting the same product list.

**Comment by jacobarriola** [comment]
@akshaykatale99 I ran into this issue a while back when trying to sort by price. You may want to look at my PR and see if that filter we've added helps with your price meta issue.

https://github.com/wp-graphql/wp-graphql/pull/1546

**Comment by akshaykatale99** [comment]
hi, @jacobarriola , thanks for the quick response. I am getting this behavior when I add categoryId to query variables. if I remove that then products are getting as expected. thanks

**Comment by gvocale** [comment]
I am running into the same issue. It seems to happen only when I am using `orderBy` as filter on a `products` query:

```gql
query Products($after: String, $before: String, $first: Int, $last: Int, $where: RootQueryToProductConnectionWhereArgs) {
  products(
    after: $after
    before: $before
    first: $first
    last: $last
    where: $where
  ) {
    nodes {
      __typename
    }
    pageInfo {
      endCursor
      hasNextPage
      hasPreviousPage
      startCursor
    }
  }
}
```
Example variables
```gql
{
  "first": 24,
  "after": "YXJyYXljb25uZWN0aW9uOjE5MTM5",
  "where": {
    "orderby": [
      {
        "field": "MENU_ORDER",
        "order": "ASC"
      }
    ],
    "status": "publish",
    "visibility": "VISIBLE"
  }
}
```

**Comment by kidunot89** [comment]
This issue was resolved in `v0.12.1`

**Comment by pooriaset** [comment]
Hi, I ran into this bug in version‍ `0.21.1` when I try to sort products by `PRICE` in ascending order. This only happens when I use `PRICE` sorting. Do you have any idea what might be causing the issue?  I’ll put the queries below.

```
query firstPage {
  products(first: 3, where: {orderby: {field: PRICE, order: DESC}}) {
    nodes {
      title
    }
    pageInfo {
      endCursor
      startCursor
    }
  }
}

query secondPage {
  products(
    first: 3
    after: "YXJyYXljb25uZWN0aW9uOjE3Mg=="
    where: {orderby: {field: PRICE, order: DESC}}
  ) {
    nodes {
      title
    }
    pageInfo {
      endCursor
      startCursor
    }
  }
}

query thirdPage {
  products(
    first: 3
    after: "YXJyYXljb25uZWN0aW9uOjQy"
    where: {orderby: {field: PRICE, order: DESC}}
  ) {
    nodes {
      title
    }
    pageInfo {
      endCursor
      startCursor
    }
  }
}
```
